snowpack/src/sources/local.ts

```diff
@@ -494,13 +495,10 @@ export class PackageSourceLocal implements PackageSource {
       const lineBullet = colors.dim(depth === 0 ? '+' : '└──'.padStart(depth * 2 + 1, ' '));
       let packageFormatted = spec + colors.dim('@' + packageVersion);
       const existingImportMapLoc = path.join(installDest, 'import-map.json');
-      const importMapHandle = await fs.open(existingImportMapLoc, 'r+').catch(() => null);
-
-      let existingImportMap: ImportMap | null = null;
-      if (importMapHandle) {
-        const importMapData = await importMapHandle.readFile('utf-8');
-        existingImportMap = importMapData ? JSON.parse(importMapData) : null;
-        await importMapHandle.close();
+      let existingImportMap: ImportMap | undefined = memoizedImportMap[packageName];
```

Comment:
This is a change to something @matthewp implemented: a race condition on reading the file system. I used this as the memoizer, because everything else is technically concerned with something else (i.e. resolution).

By putting this in-memory, this should fix the face condition (hopefully). And it’s also something we use to check (below)

Reply:
I think this still presents a race condition:

1. 2 calls to buildPackageImport for the same package occur 
1. `memoizedResolve[entrypoint]` is undefined in both cases.
1. Both try to create an export map at the same time.

To make this not race, you need to create an entry in `memoizedImportMap[packageName]` right after the check. Essentially all the code that follows should be a promise that will resolve once its written. So the check needs to await that promise if it exists.